src/solver.cpp

```diff
@@ -515,14 +516,25 @@ void Solver::applySensitivityTolerances() const {
 
 SensitivityMethod Solver::getSensitivityMethod() const { return sensi_meth; }
 
+SensitivityMethod Solver::getSensitivityMethodPreequilibration() const { return sensi_meth_preeq; }
+
 void Solver::setSensitivityMethod(const SensitivityMethod sensi_meth) {
+    if (sensi_meth != this->sensi_meth)
+        resetMutableMemory(nx(), nplist(), nquad());
+    sensitivityMethod(&this->sensi_meth, sensi_meth);
+}
+
+void Solver::setSensitivityMethodPreequilibration(const SensitivityMethod sensi_meth_preeq) {
+    sensitivityMethod(&this->sensi_meth_preeq, sensi_meth_preeq);
+}
+
+void Solver::sensitivityMethod(SensitivityMethod *member,
```

Comment:
For future reference: This implementation probably overstretches the goals of code reuse and can easily be a bit confusing. Generally, never pass the reference of a class member as argument to a function. Best solution here would be to check the value in this method and set the value outside of this function

Reply:
Well, you told me to write something reusable for both sensitivityMethod setters... ;)
This was my honest best guess... I can rewrite that in the upcoming PR, if it's helpful. No problem...

Reply:
The primary purpose of avoiding duplication is maintainability and not so much about enacting the maximum amount of abstraction possible (that usually leads to complex unreadable code). De-duplicating the value checks make sense since those checks may change and you want to have them in a central place and not change them in multiple locations. De-duplicating variable assignments doesn't really improve maintainability, as you can't programmatically access class members in C++, and passing pointers to class members to class methods is confusing as it obfuscates the fact that those pointers point to class members.